src/katsdpcontroller/tasks.py

```diff
@@ -736,6 +739,11 @@ async def unhandled_request(self, ctx, req: aiokatcp.core.Message) -> None:
         ctx.reply(aiokatcp.core.Message.OK)
         await ctx.drain()
 
+    def get_command_value(self, value_type: Callable[..., _T], parameter_string: str) -> _T:
+        """Return the value passed to the fake device as a cli parameter."""
+        position = self.logical_task.command.index(parameter_string)
+        return value_type(self.logical_task.command[position + 1])
```

Comment:
Does taking `value_type` actually add much? I don't think `get_command_value(int, "--foo")` is any more convenient than `int(get_command_value("--foo"))`. I guess it does ensure that you don't forget to convert at all.

Reply:
So without the built-in conversion, it would be something like 

```python
def get_command_argument(self, argument_name: str) -> str:
   ...
   return self.logical_task.command[position + 1]
```

Then the caller would use it like this
```
my_int_value = int(self.get_command_argument("--my-int-value"))
```

So no `Callable[[str], [_T]]` needed basically.

I've personally got no preference as to which is better, but since I've already done the one approach it's not going to save me any typing to undo it.

Reply:
I'm happy to keep it as is then.